app/Services/PirepService.php

```diff
@@ -719,4 +726,101 @@ public function setPilotState(User $pilot, Pirep $pirep)
 
         event(new UserStatsChanged($pilot, 'airport', $previous_airport));
     }
+
+    public function handleDiversion(Pirep $pirep): void
+    {
+        // Return if diversion handling is disabled
+        if (!setting('pireps.handle_diversion', false)) {
+            return;
+        }
+
+        $diversion_airport_id = $pirep->fields->where('slug', 'diversion-airport')->first()?->value;
```

Comment:
Is it better to add a field to the table for diversion airport id?

Reply:
It works without it so I don't know if it's necessary but it might make it clearer